### PR TITLE
feat(nuxt-ripple): added social links to footer

### PR DESCRIPTION
**Pull request by jeffdowdle**
**Issue**: https://digital-vic.atlassian.net/browse/R20-1507

### What I did

- Added mapping for social links, with icon mapping from ripple 1
- Improve logging of error causes in mapping functions

### How to test

- 
- 

#### For all PR's

- [ ] I've added relevant changes to the project Readme if needed.
- [ ] I've updated the documentation site as needed.
- [ ] I have added unit tests to cover my changes (if not applicable, please state why in a comment)

#### For new components only

- [ ] I have added a story covering all variants
- [ ] I have checked a11y tab in storybook passes
- [ ] Any events are emitted on the event bus

### CODE REVIEW

**Review by waitingallday — APPROVED** [review]

File: packages/ripple-tide-api/src/services/tide-api-base.ts [hunk 1]
```diff
@@ -52,16 +52,7 @@ export default class TideApiBase extends HttpClient {
...
-      return await this.getMappedDataAux(mapping, resource)
-    } catch (error) {
-      throw new ApplicationError(
-        `An error occurred while mapping tide API data`,
-        {
-          cause: error
-        }
-      )
-    }
+    return await this.getMappedDataAux(mapping, resource)
```
Comment (dylankelly) [inline packages/ripple-tide-api/src/services/tide-api-base.ts:55]: Help me understand this - does removing the try catch return the mapping field error? Where is the error returned?

File: packages/nuxt-ripple/mapping/utils/processSiteSocialLinks.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  switch (hostNormalised) {
+    case 'facebook.com':
+      return 'icon-facebook'
+    case 'instagram.com':
+    case 'instagr.am':
+      return 'icon-instagram'
+    case 'linkedin.com':
+      return 'icon-linkedin'
+    case 'twitter.com':
+      return 'icon-twitter'
```
Comment (dylankelly) [inline packages/nuxt-ripple/mapping/utils/processSiteSocialLinks.ts:26]: this will need updating for https://digital-vic.atlassian.net/browse/SPE-186

**Review by dylankelly — APPROVED** [review]

File: packages/ripple-tide-api/src/services/tide-api-base.ts (hunk 1 above)
Comment (jeffdowdle) [reply packages/ripple-tide-api/src/services/tide-api-base.ts:55]: Catching and re-throwing it here was swallowing up the actual cause of the error when it is finally logged. There's more work we can do to improve logging but this fixes a common gripe, the intermediate error doesn't add much useful information.

Before:
```
ApplicationError: Error fetching site data
   ...stack trace
Caused by: ApplicationError: An error occurred while mapping tide API data
   ...unhelpful stack trace
```

After:
```
ApplicationError: Error fetching site data
   ...stack trace
Caused by: ReferenceError: footerSection is not defined
  ...helpful stack trace
```

Comment (jeffdowdle) [reply packages/ripple-tide-api/src/services/tide-api-base.ts:55]: The error instead gets caught at the api route level like all other errors: https://github.com/dpc-sdp/ripple-framework/blob/feature/R20-1507-social-links/packages/ripple-tide-api/src/utils/createHandler.ts#L59

File: packages/nuxt-ripple/mapping/utils/processSiteSocialLinks.ts (hunk 2 above)
Comment (jeffdowdle) [reply packages/nuxt-ripple/mapping/utils/processSiteSocialLinks.ts:26]: Had that thought too, let's do it as part of a big X ticket

**Pull request merged**